src/pandablocks/asyncio.py

```diff
@@ -132,48 +132,61 @@ async def data(
         scaled: bool = True,
         flush_period: Optional[float] = None,
         frame_timeout: Optional[float] = None,
+        flush_event: Optional[asyncio.Event] = None,
     ) -> AsyncGenerator[Data, None]:
         """Connect to data port and yield data frames
 
         Args:
             scaled: Whether to scale and average data frames, reduces throughput
             flush_period: How often to flush partial data frames, None is on every
-                chunk of data from the server
+                chunk of data from the server. 0 is never and flushing will be
+                manual.
```

Comment:
Am I right in thinking that if `flush_period` is `0`, then there MUST be a `flush_event` otherwise events will simply never be flushed?

Reply:
Correct, I'll raise an error for that case